test/4.public.UrlChecker.js

```diff
@@ -139,9 +139,84 @@ describe("PUBLIC -- UrlChecker", function()
 			}).enqueue( conn.absoluteUrl );
 		});
 	});
-	
-	
-	
+
+
+
+	describe("caching", function()
+	{
+		it("should check unique url only once", function(done)
+		{
+			var options = utils.options({cacheResponses: true});
+			var results = [];
+			var success = false;
+
+			var instance = new UrlChecker( options,
+			{
+				link: function(result, customData)
+				{
+					console.log(result == undefined);
+					if (result.decorated === true) {
+						success = true;
+					}
+					result.decorated = true;
+					results[customData.index] = result;
+				},
+				end: function()
+				{
+					expect(success).to.equal(true);
+					expect(results).to.have.length(3);
+					done();
+				}
+			});
+			
+			instance.enqueue( conn.absoluteUrl+"/fixtures/index.html",     null, {index:0} );
+			instance.enqueue( conn.absoluteUrl+"/fixtures/index.html",     null, {index:1} );
+			instance.enqueue( conn.absoluteUrl+"/fixtures/link-real.html", null, {index:2} );
+		});
+
+
+
+		it("should re-check url after clearing cache", function(done)
+		{
+			var failure = false;
+			var finalFired;
+			var options = utils.options({cacheResponses: true});
+			var results = [];
+
+			var instance = new UrlChecker( options,
+			{
+				link: function(result, customData)
+				{
+					if (result.decorated === true) {
+						failure = true;
```

Comment:
instead of using an additional variable, you could do `done( new Error("this should not have happened") )`